sdk/eventgrid/azure-eventgrid/tests/unittests/test_binary_mode.py

```diff
@@ -0,0 +1,99 @@
+from azure.eventgrid._model_base import AzureJSONEncoder
+from azure.eventgrid.models import *
+from azure.core.messaging import CloudEvent
+
+class MyTestClass(object):
+    def __init__(self, name):
+        self.name = name
+    def __str__(self):
+        return self.name
+    
+class TestEGClientExceptions():
+    
+    def test_binary_request_format(self):
+        event = CloudEvent(
+            type="Contoso.Items.ItemReceived",
+            source="source",
+            subject="MySubject",
+            data=b'this is binary data',
+        )
+
+        request = _to_http_request("https://eg-topic.westus2-1.eventgrid.azure.net/api/events", event=event)
+
+        assert request.data == b"this is binary data"
+        assert request.headers.get("ce-source") == "source"
+        assert request.headers.get("ce-subject") == "MySubject"
+        assert request.headers.get("ce-type") == "Contoso.Items.ItemReceived"
+
+    
+    def test_binary_request_format_with_extensions(self):
+        event = CloudEvent(
+            type="Contoso.Items.ItemReceived",
+            source="source",
+            subject="MySubject",
+            data='this is binary data',
+            extensions={"extension1": "value1", "extension2": "value2"}
+        )
+
+        request = _to_http_request("https://eg-topic.westus2-1.eventgrid.azure.net/api/events", event=event)
+
+        assert request.data == b"this is binary data"
+        assert request.headers.get("ce-source") == "source"
+        assert request.headers.get("ce-subject") == "MySubject"
+        assert request.headers.get("ce-type") == "Contoso.Items.ItemReceived"
+
+    def test_binary_request_format_with_extensions_and_datacontenttype(self):
+        event = CloudEvent(
+            type="Contoso.Items.ItemReceived",
+            source="source",
+            subject="MySubject",
+            data='this is my data',
+            datacontenttype="application/json",
+            extensions={"extension1": "value1", "extension2": "value2"}
+        )
+
+        request = _to_http_request("https://eg-topic.westus2-1.eventgrid.azure.net/api/events", event=event)
+
+        assert request.data == b"this is my data"
+        assert request.headers.get("ce-source") == "source"
+        assert request.headers.get("ce-subject") == "MySubject"
+        assert request.headers.get("ce-type") == "Contoso.Items.ItemReceived"
+
+    def test_class_binary_request_format_error(self):
+        test_class = MyTestClass("test")
+        event = CloudEvent(
+            type="Contoso.Items.ItemReceived",
+            source="source",
+            subject="MySubject",
+            data=test_class,
+            datacontenttype="application/json",
+            extensions={"extension1": "value1", "extension2": "value2"}
+        )
+
+        with pytest.raises(TypeError):
+            _to_http_request("https://eg-topic.westus2-1.eventgrid.azure.net/api/events", event=event)
+
+    def test_xml_binary_request_format(self):
```

Comment:
is there an error scenario for passing in content_type explicitly? valid and invalid ones with binary mode